packages/thirdweb/src/chainsaw/endpoints/getNFTsByCollection.ts

```diff
@@ -0,0 +1,66 @@
+import type { ThirdwebClient } from "../../client/client.js";
+import type { Address } from "../../utils/address.js";
+import { getClientFetch } from "../../utils/fetch.js";
+import { formatChainsawNFTs } from "../formatter.js";
+import { addPagingToRequest } from "../paging.js";
+import type {
+  ChainsawPagingParams,
+  ChainsawResponse,
+  NFTs,
+  NFTsData,
+} from "../types.d.ts";
+import { getNftsByCollectionEndpoint } from "../urls.js";
+
+export type GetNFTsGroupBy = "ownerAddress";
+
+export type GetNFTsByCollectionParams = {
+  client: ThirdwebClient;
+  contractAddresses: Address[];
+  chainIds?: number[];
+  groupBy?: GetNFTsGroupBy;
+} & ChainsawPagingParams;
+
+export type GetNFTsByCollectionResult = {
```

Comment:
How does the pagination work if the data changes between fetching pages?

Reply:
for live data you probably need to use cursor based pagination but I agree that in the short term this will likely be OK